File: 24_9_Find duplicates in an array.py
class Solution:
    def duplicates(self, arr, n):

    	# code here
    	
    	ans = []
    	
    	fre = [0]*n
        for i in range(n):
            fre[arr[i]] +=1
            
        for i in range(n):
            if fre[i]>1:
                ans.append(i)
                
        if not ans:
            ans.append(-1)
        
        return ans


#{ 
 # Driver Code Starts
if(__name__=='__main__'):
    t = int(input())
    for i in range(t):
        n = int(input())
        arr = list(map(int, input().strip().split()))
        res = Solution().duplicates(arr, n)
        for i in res:
            print(i,end=" ")
        print()



# } Driver Code Ends

File: 24_9_Find duplicates in an array.cpp
class Solution{
  public:
    vector<int> duplicates(int arr[], int n) {
        // code here
        // int n = nums.size();
     unordered_map<int, int> map;

        vector<int> res;
        for(int i = 0 ; i < n; i++){
                map[arr[i]]++;
        }
        for(auto i : map){
            if(i.second > 1){
                res.push_back(i.first);
            }
        }
        if(res.size() > 0){
            sort(res.begin(), res.end());
            return res;
        }
        res.push_back(-1);
        return res;
    }
};